scripts/lib/api/generateApiComponents.ts

```diff
@@ -350,3 +399,44 @@ export async function htmlSignatureToMd(
     .replace(/^`/, "")
     .replace(/`$/, "");
 }
+
+function getHeaderLevel($: CheerioAPI, $main: Cheerio<any>, $dl: Cheerio<any>) {
+  const priorHeaderLevel = getPriorHeaderLevel($, $main, $dl);
+  if (priorHeaderLevel) {
+    if (+priorHeaderLevel == 6) {
+      throw new Error("API component cannot set inexisting header: <h7>");
+    }
+
+    return Math.max(3, +priorHeaderLevel + 1);
+  }
+
+  // Minimum header level for components without a dedicated page. This components should
+  // have an <h1> for the page and <h2> to define the section they belong to.
+  return 3;
+}
+
+function getPriorHeaderLevel(
+  $: CheerioAPI,
+  $main: Cheerio<any>,
+  $dl: Cheerio<any>,
+): string | undefined {
+  const siblings = $dl.siblings();
+  for (const sibling of siblings) {
+    const $sibling = $(sibling);
+    if ($sibling.data("header-type")) {
+      continue;
+    }
+    const tagName = $sibling.get(0)?.tagName;
+    if (tagName?.match(/h[1-6]/)) {
+      return tagName.substring(1);
+    }
+  }
+
+  // If there's no header among the siblings, we look for the prior inline class in some
+  // parent node previously set
+  return $main
+    .find("[data-header-type=class-header]")
```

Comment:
Why is this safe? Imagine this HTML

```html
<Function id="func" />

<Class>
   <Function id="method" />
</Class>
```

Because you're searching `$main`, wouldn't the first `Function` end up matching with the class after it?

Reply:
Good question! That wouldn't happen because the header for the class after it wouldn't have been created yet. To search for the previous headers, I leverage the fact that we set each header in the same order they appear on the website.

If the case were in reverse, like:
```mdx
<Class>
   <Function id="method" />
</Class>

<Function id="func" />
```

wouldn't be a problem either because we would find a header among the siblings before getting to this last return. All methods that are not in an inline class have a sibling header already in the HTML.

Reply:
Ah, makes sense. That would be great to comment